INTERNALS.md

```diff
@@ -23,75 +23,188 @@ Serves endpoints through 3 controllers:
 Controllers use a variety of models to perform, in a traditional MVC approach.
 
 
-### Watch process
+### Worker process
 
-This process is built from 4 key classes:
+This process (of which multiple instances can be run) executes a number of
+threads concurrently.
 
-- The `Watch` service regularly polls for subscriptions and creates
-  a `Receive` service for each;
-- The `Receive` service, for a given subscriptions, buffers events from a `Queue` and creates
-  `Deliver` services to send the to clients;
-- The `Deliver` service gracefully sends event batches over HTTP;
-- The `Queue` model abstracts out queue management with Redis, providing a
-  syncronous means to push and pop message from a queue.
+- A group of `ROUTEMASTER_WORKER_THREADS` threads runs the `main` job queue
+  which delivers batches.
+- A single thread executes non-delivery jobs from the `aux` queue (monitoring,
+  auto-dropping, promotion of scheduled jobs)
+- A number of ticker threads schedule auxilliary jobs.
 
-# Cron process
+Details on the services above:
 
-This process runs scheduled tasks, regularly calling the following services:
-
-- `Monitor` delivers metrics to metric adapters;
+- `Monitor` delivers gauge metrics to metric adapters;
 - `Autodrop` automatically removes the oldest messages from queues under low
   memory conditions.
 
 
+## Internal events
+
+We use the `wisper` gem as a process-local event bus. Conventionally, our event
+have a single hash as a payload. This is a catalog of such events.
+
+`events_added(name:, count:)`
+
+`events_removed(name:, count:)`
+
+
+## Event batch lifecycle
+
+All events get batched for delivery for a particular subscriber; the data model
+is described below.
+
+Batches start out _current_ when they get created. Each subscriber normally has
+at most one _current_ batch which receives new events, although in edge cases
+(high concurrency) more than one current batch might get created.
+
+When the batch is either full or stale, it gets promoted to _ready_ - another
+_current_ batch may then be created which the first awaits delivery.
+
+Finally, batches get deleted once they have been delivered; or when the
+auto-dropper removes them (because the system is running out of storage space).
```

Comment:
Yes: monitoring (which I've kept separate from this) includes auto-dropping in #50.